coalib/coala_json.py

```diff
@@ -28,12 +33,51 @@ def main():
     args = arg_parser.parse_args()
 
     log_printer = None if args.text_logs else ListLogPrinter()
+    JSONEncoder = create_json_encoder(use_relpath=args.relpath)
+
+    if args.show_bears:
+        try:
+            sections, _ = load_configuration(arg_list=None,
+                                             log_printer=log_printer)
+            local_bears, global_bears = collect_all_bears_from_sections(
+                sections, log_printer)
+            if args.filter_by_language:
+                local_bears = filter_section_bears_by_languages(
+                    local_bears, args.filter_by_language)
+                global_bears = filter_section_bears_by_languages(
+                    global_bears, args.filter_by_language)
+
+            bears = inverse_dicts(local_bears, global_bears)
+            retval = []
+            for bear, _sections in sorted(bears.items(),
+                                          key=lambda bear_tuple:
+                                          bear_tuple[0].name):
+                metadata = bear.get_metadata()
+                retval.append({"Name": bear.name,
+                               "Description": metadata.desc,
+                               "Supported languages": bear.supported_languages,
+                               "Used in": _sections,
+                               "Needed settings": [param for param in
+                                                   metadata.non_optional_params
+                                                   ],
+                               "Optional settings": [param for param in
+                                                     metadata.optional_params]
+                               })
+
+            print(json.dumps({"bears": retval},
+                             cls=JSONEncoder,
+                             sort_keys=True,
+                             indent=2,
+                             separators=(',', ': ')))
```

Comment:
naaah, I'd put this into an output variable or so, so it'll be printed to file if the user requests so